src/gencost/waterfall.py

```diff
@@ -1759,13 +1831,36 @@ def get_gf923_by_generator(self):
                     "mmbtu": "sum",
                     # "hr_src": sorted_unique_cat,
                 },
-            )
-            .reorder_levels([1, 0], axis=1)
-        )
+            ).reorder_levels([1, 0], axis=1)
 
-        gf_923.columns = map("_".join, gf_923.columns)
+            gf_923.columns = map("_".join, gf_923.columns)
+
+            return gf_923.reset_index().merge(
```

Comment:
Is it right to say that the data you are merging on here is an aggregation of data already in `gf_923`? In which case, rather than merging on this grouped data could you do the following?

```python
gf_923 = gf_923.assign(net_generation_mwh=lambda x: x.filter(like='_mwh').sum(axis=1))
```